### ISSUE TITLE
CPU number is not correctly detected

### ISSUE THREAD

**Issue by micafer** [dw/mitogen#648]
I'm getting an error when executing the kubernetes "kubeadm init" command.
This command checks the host CPU number. 
The host machine (a VM) has 2 CPUs and if mitogen is not activated the command does not raise this error. It seems that the CPU number is not correctly detected in case of using mitogen plugin.

```sh
$ ansible -i hosts node1 -m command -a "kubeadm init"

158.42.105.17_0 | FAILED | rc=1 >>
[init] Using Kubernetes version: v1.16.0
[preflight] Running pre-flight checks	[WARNING IsDockerSystemdCheck]: detected "cgroupfs" as the Docker cgroup driver. The recommended driver is "systemd". Please follow the guide at https://kubernetes.io/docs/setup/cri/
error execution phase preflight: [preflight] Some fatal errors occurred:
	[ERROR NumCPU]: the number of available CPUs 1 is less than the required 2
[preflight] If you know what you are doing, you can make a check non-fatal with `--ignore-preflight-errors=...`
To see the stack trace of this error execute with --v=5 or highernon-zero return code
```
Tested with Ansible versions 2.6.15 and 2.8.5.
Mitogen version 0.2.8
OS: Ubuntu 18.04.3
Python version 2.7.15+

**Comment by dw** [comment]
This is a curious one! Is `node1` the controller machine, and/or is the `local` connection plug-in used to reach it?

Go is detecting CPU count [by reading the inherited process CPU affinity mask](https://github.com/golang/go/blob/3b6216ed0601c81fe42c2a4738d419afccb62163/src/runtime/os_linux.go#L84-L109), which should have been reset by Mitogen. But this only applies to the controller machine -- targets should be unaffected.

Can you please supply a `-vvv` log for your run above, it might have some useful warnings hiding in the output. Thanks for reporting this!

**Comment by dw** [comment]
Looks like the affinity is somehow not being reset correctly with the `local` connection plug-in:

````
[13:19:20 eldil!7 ansible] ANSIBLE_STRATEGY=mitogen ansible target -m shell -a 'taskset -c -p $$' 
 [WARNING]: Invalid characters were found in group names but not replaced, use -vvvv to see details

target | CHANGED | rc=0 >>
pid 23917's current affinity list: 0-7

[13:19:35 eldil!8 ansible] ANSIBLE_STRATEGY=mitogen ansible target -m shell -a 'taskset -c -p $$' -c local
 [WARNING]: Invalid characters were found in group names but not replaced, use -vvvv to see details

target | CHANGED | rc=0 >>
pid 23972's current affinity list: 2-7
````

**Comment by dw** [comment]
I'm guessing for your machine, the `taskset` command is returning something like '1-1`. Would it be possible to include a copy of `/proc/cpuinfo` too? Thanks

**Comment by micafer** [comment]
> This is a curious one! Is `node1` the controller machine, and/or is the `local` connection plug-in used to reach it?

Yes, it is the controller machine using the local connection plugin.

> 
> Go is detecting CPU count [by reading the inherited process CPU affinity mask](https://github.com/golang/go/blob/3b6216ed0601c81fe42c2a4738d419afccb62163/src/runtime/os_linux.go#L84-L109), which should have been reset by Mitogen. But this only applies to the controller machine -- targets should be unaffected.
> 
> Can you please supply a `-vvv` log for your run above, it might have some useful warnings hiding in the output. Thanks for reporting this!

This is the output:
[ansible.log](https://github.com/dw/mitogen/files/3631589/ansible.log)

**Comment by micafer** [comment]
> I'm guessing for your machine, the `taskset` command is returning something like `1-1`. Would it be possible to include a copy of `/proc/cpuinfo` too? Thanks

[cpuinfo.txt](https://github.com/dw/mitogen/files/3634556/cpuinfo.txt)

**Comment by Sephtex** [comment]
We faced exactly the same issue where we detected that a VM that was shared for production deployments started to be really slow. We noticed that even when we increased the amount of cores from 4 to 8, nothing really changed, and a suspicious load was put on core nr2 (when starting from nr1). the VM got even loads of 30-40 when 5-10 playbooks where executed at the same time due to the fact that core nr.2 got 100% loaded with a lot of threads just consuming 1-3% CPU.

We have some playbooks with a lot of connection=local for local powershell and k8s executions for example.

When I removed mitogen, I instantly noticed that the load was nicely spread over all cores.
